File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.kt
class Solution {
    fun solution(babbling: Array<String>): Int {
        var answer = 0

        for(word in babbling) {
            val sb = StringBuilder()
            var previousWord = ""

            for(i in word.indices) {
                sb.append(word[i])
                val str = sb.toString()

                if(previousWord == str || str.length > 3) {
                    break
                }else if(str == "ye" || str == "ma" || str == "aya" || str == "woo") {
                    previousWord = str
                    sb.clear()
                }

                if(i == word.length-1 && sb.isEmpty()) answer++
            }
        }
        return answer
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.swift
func solution(_ babbling:[String]) -> Int {
    var answer = 0
    var pronunciations = ["aya", "ye", "woo", "ma"]
    
    for babble in babbling {
        var previous = ""
        var word = ""
        
        for b in babble {
            word += String(b)
            
            if pronunciations.contains(word) && previous != word {
                previous = word
                word = ""
            }
        }
        
        answer += (word == "") ? 1 : 0
    }
    
    return answer
}